### ISSUE TITLE
Incorrect error handling in Bitmex

### ISSUE THREAD

**Issue by StanleyDing** [ccxt/ccxt#4185]
- OS: macOS 10.14.1
- Programming Language version: python 3.7.0
- CCXT version: 1.17.507
- Exchange: Bitmex
- Method: fetchBalance()  **// don't think it matters**

When API keys are invalid, Bitmex returns with http status `401` and response
```js
{
  "error": {
    "message": "Invalid API Key.",
    "name": "HTTPError"
  }
}
```

With invalid API keys, ccxt at version 1.17.507 throws `ExchangeError` instead of `AuthenticationError`. 

I think the bug is introduced in commit 2af3fcac, where http `code` is used to check against `this.exceptions.exact`. The correct check would be to use `error.message` from the response body.

I would be more than happy to send a PR if this is indeed a bug, just let me know!

**Comment by kroitor** [comment]
@StanleyDing Yep, looks like a bug to me. Thx for reporting!

> I would more than happy to send a PR if this is indeed a bug, just let me know!

We welcome all PRs and contributions! So, yes, if you can, please, do a PR.

One important thing to note: it should be an easy one-liner fix (i'm sure that's obvious to you), however, due to our multilanguage nature, we should follow the rules from the [CONTRIBUTING doc](https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md) very precisely (to a word). Namely, the following sections of it:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-contribute-code
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#multilanguage-support
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes

Thx again, respect!

**Comment by kroitor** [comment]
It is now fixed in CCXT 1.17.509 as this was an easy one ;)

@StanleyDing Your contributions and PRs are welcome anyways, there's plenty of room for development! Hope to hear from you in future )
